application/common/components/MfaBackendInterface.php

```diff
@@ -9,9 +9,10 @@ interface MfaBackendInterface
      * @param int $userId
      * @param string $mfaExternalUuid (only used for WebAuthn)
      * @param string $rpOrigin Relying Party Origin (only used for WebAuthn)
+     * @param string $recoveryEmail for mfa account recovery (only used for WebAuthn)
      * @return array
      */
-    public function regInit(int $userId, string $mfaExternalUuid, string $rpOrigin): array;
+    public function regInit(int $userId, string $mfaExternalUuid, string $rpOrigin, ?string $recoveryEmail = null): array;
```

Comment:
This is possibly too nit-picky, but it may be better to follow the example of the `$rpOrigin` parameter, and default to an empty string instead of making `$recoveryEmail` a nullable string:
```suggestion
    public function regInit(int $userId, string $mfaExternalUuid, string $rpOrigin, string $recoveryEmail = ''): array;
```

Reply:
I don't have a strong preference either way. We did it this way because the getParam can return null, but that is easily fixable.